Add readme

dnl-bsch opened the pull request.

Commit:
add first version of readme

Commit:
add first version of README

Commit:
[DATALAD] Recorded changes

Commit:
[DATALAD] Recorded changes

Commit:
[DATALAD] Recorded changes

dnl-bsch commented:
@znichollscr can you take a look

Commit:
[DATALAD] added README

Commit:
adding parts from UNFCCC repo and automatically created DataLad README, DataLad sibling description, merge instructions.

znichollscr commented on README.md:
```diff
@@ -0,0 +1,200 @@
+## Installation
+
+- Install datalad according to the [DataLad handbook](https://handbook.datalad.org/en/latest/intro/installation.html). It is recommended to install globally. 
+- DataLad is based on Git. [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) needs to be installed to run DataLad. 
```
```suggestion
```

I think we can delete this. I suspect datalad will install git if it isn't already there (deleting the line just means one less thing for us to worry about)

znichollscr commented on README.md:
```diff
@@ -0,0 +1,200 @@
+files, you can find out what has been done to the dataset or to individual  
+files by whom, and when.
+
+## Executing the program
```
```suggestion
## Contributing

For those who wish to contribute to the repository, below we go through the key commands you will need to use. 
```

Just something to give a bit of a context so people know what the program is and what you would use it for.

znichollscr commented on README.md:
````diff
@@ -0,0 +1,200 @@
+doit setup_env
+```
+#### <a name="download"></a>Download the version from the command line.
+This will download all files from Zenodo as they are.
````
```suggestion
This will download all files from Zenodo as they are for a specific version (note this version must already be in `versions.py`, if you want to add a new version, see the section on adding a new version below).
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,200 @@
+
+
+To add a new version go to **versions.py** in the **src** directory and create a new value in the
+dictionary. Fill all the required information similar to the previous entries.
```
```suggestion
`versions` dictionary. Fill all the required information similar to the previous entries.
```

znichollscr commented on README.md:
`````diff
@@ -0,0 +1,200 @@
+    },
+}
+````
+Then run the two commands `read_version` and `download_version` as described in **Executing the program**. 
`````
```suggestion
Then run the two commands `read_version` and `download_version` as described in ****. 
```

Update name

JGuetschow replied:
Do you mean to replace **Executing the program** by nothing?

znichollscr commented on README.md:
````diff
@@ -0,0 +1,200 @@
+**Push to the github repository**
+```
+datalad push --to <name>
+
````
```suggestion
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,200 @@
+pure git repositories. As the data is only stored on gin, the gin repository is the source to start 
+from. As gin currently has a problem with forks (the annexed data is not 
+forked) we have to use branches for development and, thus, to contribute you
+first need to contact the maintainers to get write access to the repository.
```
```suggestion
first need to contact the maintainers to get write access to the gin repository.
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,200 @@
+### Instructions for merge requests
+Once you have everything set up you can create a new branch branch and work there. 
+When you're done create a pull request to integrate your work into the main 
+branch. This should be done first github to allow for discussions and review. Afterwards the changes 
```
```suggestion
branch. This should be done first on github to allow for discussions and review (gin servers don't have the same review features). Afterwards the changes 
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,200 @@
+Once you have everything set up you can create a new branch branch and work there. 
+When you're done create a pull request to integrate your work into the main 
+branch. This should be done first github to allow for discussions and review. Afterwards the changes 
+can be implemented on gin. 
```
```suggestion
can be actually merged on gin (so that the annex is merged properly too). 
```

znichollscr reviewed (commented):
Super nice. Let's discuss a bit tomorrow, maybe clean up a touch more then we can merge and use this as a basis for a copier repository (from which we can then re-create this README to make everything consistent, you'll be a copier pro in no time).

JGuetschow commented on .gitignore:
```diff
@@ -1,6 +1,7 @@
 __pycache__
 venv
-.doit.db
+.doit.db.*
```
```suggestion
.doit.db.*
.doit.db
```
The old entry has to stay else it will not ignore the original file. I don't have .doit.db.db

JGuetschow commented on README.md:
```diff
@@ -0,0 +1,200 @@
+# Global CO2 from Cement Production Dataset
+
+This repository downloads the Andrew dataset on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/10008931).
+
```
```suggestion
The dataset is converted to the PRIMAP2 format and provided in the csv based interchange format and the netCDF based native primap2 format. Several version of the dataset are available.

```

JGuetschow commented on README.md:
```diff
@@ -0,0 +1,200 @@
+# Global CO2 from Cement Production Dataset
+
+This repository downloads the Andrew dataset on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/10008931).
```
```suggestion
This repository downloads the Andrew dataset on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/831454).
```
I replaced the DOI by the DOI for all versions.

JGuetschow commented on README.md:
```diff
@@ -0,0 +1,200 @@
+
+## Description
+
+This repository downloads data on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/10008931).
```
```suggestion
This repository downloads data on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/831454).
```

JGuetschow commented on README.md:
```diff
@@ -0,0 +1,200 @@
+## Description
+
+This repository downloads data on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/10008931).
+The downloaded dataset can then be converted into CSV (.csv file extension) or NetCDF (.nc file extension) format.
```
```suggestion
The downloaded dataset can then be converted into CSV (.csv file extension) or NetCDF (.nc file extension) format. Converted data are available for the following versions:
| v231016 |[Zenodo](https://zenodo.org/records/10008931) |
| v230913 |[Zenodo](https://zenodo.org/records/8339353) |
| v230428 |[Zenodo](https://zenodo.org/records/7875557) |
| v220915 |[Zenodo](https://zenodo.org/records/7081360) |
| v220516 |[Zenodo](https://zenodo.org/records/6553090) |
```

JGuetschow replied:
I think it's good to have the versions that are available somewhere. And the info that you don't need to convert anything. The table is not displayed properly by github. Not sure if it will work in the README.

JGuetschow commented on README.md:
```diff
@@ -0,0 +1,200 @@
+- Install datalad according to the [DataLad handbook](https://handbook.datalad.org/en/latest/intro/installation.html). It is recommended to install globally. 
+- DataLad is based on Git. [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) needs to be installed to run DataLad. 
+- Install [Python](https://www.python.org)
+- [pydoit](https://pydoit.org/install.html)
```
Python and pydoit are only needed if you actually run the scripts. if you just want the data you don't need them (though this case is likely very rare to non-existent as the data is meant to be used in a python based tool and already publicly available for general use cases)

JGuetschow reviewed (commented):
Looks great. Lot's of info that can be used for many repositories. I've also added a few comments

dnl-bsch committed:
Update README.md

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

dnl-bsch committed:
Update README.md

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

dnl-bsch committed:
Update README.md

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

dnl-bsch committed:
Update README.md

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

dnl-bsch committed:
Update README.md

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

dnl-bsch committed:
Update README.md

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

dnl-bsch committed:
Update README.md

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

dnl-bsch committed:
Update README.md

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

dnl-bsch committed:
Update README.md

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

dnl-bsch committed:
Update .gitignore

Co-authored-by: Johannes Gütschow <mail@johannes-guetschow.de>

dnl-bsch committed:
Update README.md

Co-authored-by: Johannes Gütschow <mail@johannes-guetschow.de>

dnl-bsch committed:
Update README.md

Co-authored-by: Johannes Gütschow <mail@johannes-guetschow.de>

dnl-bsch committed:
Update README.md

Co-authored-by: Johannes Gütschow <mail@johannes-guetschow.de>

dnl-bsch committed:
Update README.md

Co-authored-by: Johannes Gütschow <mail@johannes-guetschow.de>

Commit:
test

znichollscr commented on README.md:
```diff
@@ -0,0 +1,207 @@
+This repository downloads the Andrew dataset on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/831454).
+The dataset is converted to the PRIMAP2 format and provided in the csv based interchange format and the netCDF based native primap2 format. Several version of the dataset are available.
+
+test
```
```suggestion
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,207 @@
+## Description
+
+This repository downloads data on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/831454).
+The downloaded dataset can then be converted into CSV (.csv file extension) or NetCDF (.nc file extension) format. Converted data are available for the following versions:
+| v231016 |[Zenodo](https://zenodo.org/records/10008931) |
```
```suggestion
The downloaded dataset can then be converted into CSV (.csv file extension) or NetCDF (.nc file extension) format. Converted data are available for the following versions:

| v231016 |[Zenodo](https://zenodo.org/records/10008931) |
```

znichollscr reviewed (commented):
Quick test example

dnl-bsch committed:
Apply suggestions from code review

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

Commit:
Merge remote-tracking branch 'origin/main' into add-readme

Commit:
small changes to readme

dnl-bsch commented:
@JGuetschow @znichollscr thanks for the suggestions. I went through the Readme again and made some small changes. Can you take a look?

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+# Global CO2 from Cement Production Dataset
+
+This repository downloads the Andrew dataset on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/831454).
```
```suggestion
This repository downloads the Andrews dataset on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/831454).
```

I think surname is Andrews not Andrew (confusingly, Andrews is their surname not first name)

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+# Global CO2 from Cement Production Dataset
+
+This repository downloads the Andrew dataset on global CO2 emissions from cement production from [Zenodo](https://zenodo.org/records/831454).
+The dataset is converted to the PRIMAP2 format and provided in the csv based interchange format and the netCDF based native primap2 format. Several version of the dataset are available.
```
```suggestion
The dataset is converted to the PRIMAP2 format and provided in both the csv-based interchange format and the netCDF-based native primap2 format. Several version of the dataset are available within this repository.
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+| v230913 |[Zenodo](https://zenodo.org/records/8339353) |
+| v230428 |[Zenodo](https://zenodo.org/records/7875557) |
+| v220915 |[Zenodo](https://zenodo.org/records/7081360) |
+| v220516 |[Zenodo](https://zenodo.org/records/6553090) |
```
```suggestion
- | v231016 |[Zenodo](https://zenodo.org/records/10008931) |
- | v230913 |[Zenodo](https://zenodo.org/records/8339353) |
- | v230428 |[Zenodo](https://zenodo.org/records/7875557) |
- | v220915 |[Zenodo](https://zenodo.org/records/7081360) |
- | v220516 |[Zenodo](https://zenodo.org/records/6553090) |

```

If you put a hyphen at the start then this will show as dot points which is a bit nicer to read

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+| v220915 |[Zenodo](https://zenodo.org/records/7081360) |
+| v220516 |[Zenodo](https://zenodo.org/records/6553090) |
+The data management tool [DataLad](http://docs.datalad.org/en/stable/) is used to version control the data sets.
+Commands to run the scripts are executed via the pydoit package.
```
```suggestion
Commands to manage the data are executed via the [pydoit](https://pydoit.org/) package.
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+fine-grained data access down to the level of individual files, and allows for  
+tracking future updates. In order to use this repository for data retrieval,  
+[DataLad](https://www.datalad.org/) is required. It is a free and open source  
+command line tool, available for all major operating systems, and builds up on  
```
```suggestion
command line tool, available for all major operating systems, and builds upon  
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+[DataLad](https://www.datalad.org/) is required. It is a free and open source  
+command line tool, available for all major operating systems, and builds up on  
+Git and [git-annex](https://git-annex.branchable.com/) to allow sharing,  
+synchronizing, and version controlling collections of large files.  
```
Out of curiosity, did this come out of `datalad readme` or did you write this? Very good either way

dnl-bsch replied:
That's datalad readme

znichollscr replied:
Nice

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+
+## Installation
+
+- Install datalad according to the [DataLad handbook](https://handbook.datalad.org/en/latest/intro/installation.html). It is recommended to install globally. 
```
```suggestion
- Install datalad according to the [DataLad handbook](https://handbook.datalad.org/en/latest/intro/installation.html). We recommend installing datalad globally as managing it from within the venv is not something we do for you.
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+
+- Install datalad according to the [DataLad handbook](https://handbook.datalad.org/en/latest/intro/installation.html). It is recommended to install globally. 
+- Install [Python](https://www.python.org)
+- [pydoit](https://pydoit.org/install.html)
```
```suggestion
- Install [pydoit](https://pydoit.org/install.html). Like with datalad, we recommend installing pydoit globally as managing it from within the venv is not something we do for you.
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+- Install [Python](https://www.python.org)
+- [pydoit](https://pydoit.org/install.html)
+
+Note that for [simply downloading the dataset,](#1-easy-access) Python and pydoit are not required.
```
Maybe move this up to the start of this section so people don't accidentally install that stuff before they get to this point

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+datalad clone
+```
+Do not use **git clone** to download the repository! This way DataLad will not have the necessary
+information to run the program. Once a dataset is cloned, it is a light-weight directory on your local machine.  
````
```suggestion
information to manage the dataset. Once the repository is cloned, it is like using a standard light-weight repository on your local machine.  
```

Unless this is auto-generated from datalad and I am using the wrong language...?

dnl-bsch replied:
Yes this is auto-generated. I agree that "manage dataset" describes better what DataLad actually does than  "run program". For "directory" and "repository" I don't really know what the difference is. A quick Google search says: "A directory is a location in a file system, like /bin. A repository is typically a bunch of version controlled files." So repository would be the better description.

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+Do not use **git clone** to download the repository! This way DataLad will not have the necessary
+information to run the program. Once a dataset is cloned, it is a light-weight directory on your local machine.  
+At this point, it contains only small metadata and information on the identity  
````
```suggestion
At this point, the repository contains only small metadata and information on the identity  
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+Do not use **git clone** to download the repository! This way DataLad will not have the necessary
+information to run the program. Once a dataset is cloned, it is a light-weight directory on your local machine.  
+At this point, it contains only small metadata and information on the identity  
+of the files in the dataset, but not actual *content* of the (sometimes large)  
```
```suggestion
of the files in the dataset, but not the actual *content* of the (sometimes large)  
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+information to run the program. Once a dataset is cloned, it is a light-weight directory on your local machine.  
+At this point, it contains only small metadata and information on the identity  
+of the files in the dataset, but not actual *content* of the (sometimes large)  
+data files.  
```
```suggestion
files.  
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+### Stay up-to-date  
+  
+DataLad datasets can be updated. The command `datalad update` will *fetch*  
+updates and store them on a different branch (by default  
```
```suggestion
updates and store them on a different branch to the one you're currently working on (by default  
```

If I understand correctly?

dnl-bsch replied:
I guess so. This is taken from the automatically generated README by DataLad

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+datalad update --merge  
+```  
+  
+will *pull* available updates and integrate them in one go.  
````
```suggestion
will *fetch* available updates and integrate them in one go.  
```

Typically pull = fetch + merge so I think fetch is the right word here

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+
+### Clone the repository
+
+A DataLad dataset can be `cloned` by running  
```
```suggestion
A DataLad dataset can be `cloned` by running  

```

new lines before code blocks are a convention (don't ask me why)

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+### Clone the repository
+
+A DataLad dataset can be `cloned` by running  
+```
````
```suggestion
```sh
```

adding sh after ``` makes the code render a bit nicer in most viewers and also tells the user what language this code is in (in this case it's just basic shell commands)

dnl-bsch replied:
Good to know, thanks :)

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+datalad clone
+```
+Do not use **git clone** to download the repository! This way DataLad will not have the necessary
````
```suggestion

Do not use git clone to download the repository! If you use plain git clone, DataLad will not have the necessary
```

New line after code block is convention.

Language changes make things a bit clearer I think

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+Users who simply want to retrieve the dataset have the option to access both the
+original and extracted files with
+
+```
````
```suggestion
```sh
```

As above

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+dataland get <filename>
+```
+This command will trigger a download of the files, directories, or subdatasets  
````
```suggestion

This command will trigger a download of the files, directories, or subdatasets  
```

As above

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+you have specified.
+
+For example, the CSV file for the 2023/09/13 release can be downloaded with
+```
````
```suggestion
```sh
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+This command will trigger a download of the files, directories, or subdatasets  
+you have specified.
+
+For example, the CSV file for the 2023/09/13 release can be downloaded with
```
```suggestion
For example, the CSV file for the 2023/09/13 release can be downloaded with

```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+datalad get extracted_data/v230913/Robbie_Andrew_Cement_Production_CO2_230913.csv 
+```
+### Stay up-to-date  
````
```suggestion

### Stay up-to-date  
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+  
+### Find out what has been done  
+  
+DataLad datasets contain their history in the ``git log``.  By running ``git  
```
```suggestion
DataLad datasets contain their history in the `git log`.  By running `git  
```

Single back-ticks for code in markdown (double back-ticks are for rst)

dnl-bsch replied:
What's rst?

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+### Find out what has been done  
+  
+DataLad datasets contain their history in the ``git log``.  By running ``git  
+log`` (or a tool that displays Git history) in the dataset or on specific  
```
```suggestion
log` (or a tool that displays Git history) in the dataset or on specific  
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+
+For those who wish to contribute to the repository, below we go through the key commands you will need to use. 
+
+#### Set up the virtual environment with doit
```
```suggestion
#### Set up the virtual environment with doit

```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+For those who wish to contribute to the repository, below we go through the key commands you will need to use. 
+
+#### Set up the virtual environment with doit
+```
````
```suggestion
```sh
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+doit setup_env
+```
+#### <a name="download"></a>Download the version from the command line.
````
```suggestion

#### <a name="download"></a>Download the version from the command line.
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+doit setup_env
+```
+#### <a name="download"></a>Download the version from the command line.
+This will download all files from Zenodo as they are for a specific version (note this version must already be in `versions.py`, if you want to add a new version, see the section on adding a new version below).
````
```suggestion
This will download all files from Zenodo as they are for a specific version (note this version must already be in `versions.py`, if you want to add a new version, see the section on adding a new version below).

```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+#### <a name="download"></a>Download the version from the command line.
+This will download all files from Zenodo as they are for a specific version (note this version must already be in `versions.py`, if you want to add a new version, see the section on adding a new version below).
+```
````
```suggestion
```sh
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+doit download_version --version <YYMMDD>
+```
+#### <a name="convert"></a>Convert the data sets into CSV and NetCDF files.
````
```suggestion

#### <a name="convert"></a>Convert the data sets into CSV and NetCDF files.
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+This will download all files from Zenodo as they are for a specific version (note this version must already be in `versions.py`, if you want to add a new version, see the section on adding a new version below).
+```
+doit download_version --version <YYMMDD>
+```
````
Let's add an example below here to show the format exactly e.g. 

```sh
doit download_version --version v220916
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+doit download_version --version <YYMMDD>
+```
+#### <a name="convert"></a>Convert the data sets into CSV and NetCDF files.
+```
````
```suggestion

```sh
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+doit read_version --version <YYMMDD>
+```
+
````
Add example here

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+## <a name="newversion"></a> How to add a new version
+
+
+To add a new version go to **versions.py** in the **src** directory and create a new value in the
```
```suggestion
To add a new version go to `src/versions.py` in the `src` directory and create a new value in the
```

Convention is to use back-ticks around filenames and directories so they render as code rather than bold

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+
+To add a new version go to **versions.py** in the **src** directory and create a new value in the
+`versions` dictionary. Fill all the required information similar to the previous entries.
+For example, the value _v230913_ in the _versions_ dictionary describes the 13-Sep-2023 release.
```
```suggestion
For example, the value for key `"v230913"` in the `versions` dictionary describes the 13-Sep-2023 release.
```

znichollscr commented on README.md:
`````diff
@@ -0,0 +1,210 @@
+    },
+}
+````
+Then run the two commands `read_version` and `download_version` as described in [Contributing](#2-contributing). 
`````
```suggestion

Then run the two commands, `read_version` and `download_version` as described in [Contributing](#2-contributing) for your newly added version. 
```

znichollscr commented on README.md:
`````diff
@@ -0,0 +1,210 @@
+````
+Then run the two commands `read_version` and `download_version` as described in [Contributing](#2-contributing). 
+
+## Help
`````
```suggestion
## Help

```

Just convention to have newline after header (not super important)

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+
+## Help
+Show all doit commands
+```
````
```suggestion

```sh
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+Show all doit commands
+```
+doit help
+```
````
```suggestion
```

```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+doit help
+```
+See a list with possible doit commands specific to this repository
+```
````
```suggestion

```sh
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+doit help <command>
+```
+
+## Contributing
````
```suggestion
```

Already have contributing heading above so can delete here

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+
+## Contributing
+### Repository structure
+- **.datalad/** contains config file for datalad
```
```suggestion
- `.datalad/` contains config file for datalad
```

Same for all lines below so these elements appear code-like rather than bold

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+## Contributing
+### Repository structure
+- **.datalad/** contains config file for datalad
+- **downloaded_data/** contains original data from Zenodo.
+- **extracted_data/** contains data in .csv and .nc format
+- **literature/** contains link to publication by Robbie M. Andrew. Can be downloaded with _datalad get_ command
```
```suggestion
- `downloaded_data/` contains original data from Zenodo.
- `extracted_data/` contains data in .csv and .nc format
- `literature/` contains link to publication by Robbie M. Andrew. Can be downloaded with _datalad get_ command
```

etc. for the rest of this section

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+- **setup.cfg** requirements
+- **setup.py** installs python packages
+
+### Make sure to correctly set up the DataLad siblings
+Git repositories can configure clones of a dataset as _remotes_ in order to fetch, pull, or push from and to them. A `datalad sibling` is the equivalent of a git clone that is configured as a remote. 
```
```suggestion
### Make sure to correctly set up the DataLad siblings

Git repositories can configure clones of a dataset as remotes in order to fetch, pull, or push from and to them. A `datalad sibling` is the equivalent of a git clone that is configured as a remote. 
```

Is this how datalad describes siblings? The language seemed a bit odd to me but maybe this is the best description

dnl-bsch replied:
Yes that's datalad's description. I don't really know how to describe it differently. Took it from here: https://handbook.datalad.org/en/latest/basics/101-121-siblings.html#

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+### Make sure to correctly set up the DataLad siblings
+Git repositories can configure clones of a dataset as _remotes_ in order to fetch, pull, or push from and to them. A `datalad sibling` is the equivalent of a git clone that is configured as a remote. 
+
+**Query information** about about all known siblings with
```
```suggestion
**Query information** about about all known siblings with

```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+Git repositories can configure clones of a dataset as _remotes_ in order to fetch, pull, or push from and to them. A `datalad sibling` is the equivalent of a git clone that is configured as a remote. 
+
+**Query information** about about all known siblings with
+```
````
```suggestion
```sh
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+datalad siblings add --dataset . --name <name> --url git@github.com:JGuetschow/Global_CO2_from_cement_production.git
+```
+SSH-access is needed to run this command. Note that _name_ can be freely chosen. 
````
```suggestion

SSH-access is needed to run this command. Note that `name` can be freely chosen (we tend to just use "github" for GitHub siblings) 
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+
+**Add a sibling** to allow pushing to github
+```
````
```suggestion

```sh
```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+SSH-access is needed to run this command. Note that _name_ can be freely chosen. 
+
+**Push to the github repository**
````
```suggestion
**Push to the github repository**

```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+```
+datalad push --to <name>
+```
+
````
```suggestion

where `name` should match the name you used above
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+There always issues open regarding coding, some of them easy to resolve, some harder.
+
+### Your ideas
+Contributing is ouf course not limited to the categories above. I you have ideas for improvements just open an issue or a discussion page to discuss you idea with the community.
```
```suggestion
Contributing is ouf course not limited to the categories above. If you have ideas for improvements just open an issue or a discussion page to discuss your idea with the community.
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+### Issues
+There always issues open regarding coding, some of them easy to resolve, some harder.
+
+### Your ideas
```
```suggestion
### Your ideas

```

znichollscr commented on README.md:
````diff
@@ -0,0 +1,210 @@
+datalad push --to <name>
+```
+
+### Issues
````
```suggestion
### Issues

```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+For that you first need to store your public ssh key on the gin server 
+(settings -> SSH Keys). 
+
+### Instructions for merge requests
```
```suggestion
### Instructions for merge requests

```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+### Your ideas
+Contributing is ouf course not limited to the categories above. I you have ideas for improvements just open an issue or a discussion page to discuss you idea with the community.
+
+### Technical HowTo for contributors
```
```suggestion
### Technical HowTo for contributors

```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+Contributing is ouf course not limited to the categories above. I you have ideas for improvements just open an issue or a discussion page to discuss you idea with the community.
+
+### Technical HowTo for contributors
+As we have a datalad repository using github and gin the process of contributing code and data is a bit different from 
```
```suggestion
As we have a datalad repository using github and gin, the process of contributing code and data is a bit different from 
```

znichollscr commented on README.md:
```diff
@@ -0,0 +1,210 @@
+
+### Instructions for merge requests
+Once you have everything set up you can create a new branch branch and work there. 
+When you're done create a pull request to integrate your work into the main 
```
```suggestion
When you're done, create a pull request to integrate your work into the main 
```

znichollscr reviewed (approved):
Nice work. There are way more comments and nits than normal because I'm being nit picky because we'll likely re-use this or something like it. Having lots of comments is good practice for you to get used to MRs and reviews, but don't worry you'll probably never have one like this again. (For reference, in future for something like this I would probably just make a MR into your branch to save you clicking so much, but as I said good practice here.)

Nice job. Over to you to accept changes, make other changes, discuss in threads etc. then assign the MR to Johannes for final review and merging.

Commit:
[DATALAD RUNCMD] Read data for v230913.

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "./venv/bin/python3 src/read_version.py --version v230913",
 "dsid": "24f90b12-e4a9-4e2c-995d-a54ed4cd49c7",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [
  "downloaded_data/v230913/0. GCP-CEM.csv"
 ],
 "outputs": [
  "extracted_data/v230913/Robbie_Andrew_Cement_Production_CO2_230913.nc",
  "extracted_data/v230913/Robbie_Andrew_Cement_Production_CO2_230913.yaml",
  "extracted_data/v230913/Robbie_Andrew_Cement_Production_CO2_230913.csv"
 ],
 "pwd": "."
}
^^^ Do not change lines above ^^^

Commit:
[DATALAD RUNCMD] Read data for v230428.

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "./venv/bin/python3 src/read_version.py --version v230428",
 "dsid": "24f90b12-e4a9-4e2c-995d-a54ed4cd49c7",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [
  "downloaded_data/v230428/1. Cement_emissions_data.csv"
 ],
 "outputs": [
  "extracted_data/v230428/Robbie_Andrew_Cement_Production_CO2_230428.nc",
  "extracted_data/v230428/Robbie_Andrew_Cement_Production_CO2_230428.yaml",
  "extracted_data/v230428/Robbie_Andrew_Cement_Production_CO2_230428.csv"
 ],
 "pwd": "."
}
^^^ Do not change lines above ^^^

dnl-bsch committed:
Apply suggestions from code review

Co-authored-by: znichollscr <114576287+znichollscr@users.noreply.github.com>

Commit:
formatting and small additions

dnl-bsch commented:
Hi, thank you for your comments. I did the changes in the Readme. I hope I caught all the comments. Let me know if I should change anything else. @znichollscr

znichollscr commented:
All looks good to me. We've made a bit of a mess here, so I would suggest we do a very simple, clean commit on Gin. I'll ping you on slack
